### PR TITLE
tests: remove `rev` for POSIX compatibility

### PR DESCRIPTION
**Pull request by gperciva**

### CODE REVIEW

**Comment by cperciva** [comment]
Are you sure this can't be achieved just with shell parameter expansion?  e.g. to remove a suffix ${foo%%${bar}}

**Comment by gperciva** [comment]
Do you dislike the `basename`, or the `tail`?

Ultimately, we want to go from

    /data/td/src/libcperciva/tests-output/01-crc32-00.exit

to

    /data/td/src/libcperciva/tests-valgrind/01-crc32-00

A different option is to do

    base_with_num=$(basename "${exitfile}" ".exit")
    echo "${out_valgrind}/${base_with_num}"

Of course, we could use `sed` and `tail`, or `sed` and shell parameter expansion, or...?

**Comment by gperciva** [comment]
I did double-check that using `basename` to remove a suffix is in POSIX:
https://pubs.opengroup.org/onlinepubs/9699919799/utilities/basename.html

**Comment by cperciva** [comment]
It was the `tail -c -3` which felt most awkward to me.  I'd prefer your alternative of using `${out_valgrind}`; alternatively, I think

```
basefile=${exitfile%%.exit}
echo "${val_basename}-${basefile##*-}"
```

would work, I think?  (And the string manipulation would be entirely within `sh`.)

**Comment by gperciva** [comment]
Updated PR, checked on linux (spiped, for valgrind on multiple files) and solaris.

File: tests/shared_test_functions.sh [hunk 1]
```diff
@@ -269,8 +269,8 @@ setup_check_variables() {
 get_val_basename() {
 	val_basename=$1
```
Comment (cperciva) [inline tests/shared_test_functions.sh:270]: Looks like `val_basename` is not being used now?

Comment (gperciva) [reply tests/shared_test_functions.sh:270]: Huh.  I'll revisit this later.

**Comment by gperciva** [comment]
Updated, ready for merge, checked on freebsd, linux, and solaris.

**Comment by cperciva** [comment]
Am I right in assuming that `out_valgrind` is a global variable which we can assume is set appropriately?

And did you mean to remove the "draft" label?

**Comment by gperciva** [comment]
Yes, currently we assume that `${out_valgrind}` is set (for example, in `prepare_directories()`).  I should probably check that explicitly, but this PR does not change the assumption-ness of `${out_valgrind}`.

**Pull request merged**